source/server/async_server_write_queue.h

```diff
@@ -0,0 +1,58 @@
+#ifndef NIDEVICE_GRPC_DEVICE_ASYNC_SERVER_WRITE_QUEUE_H
+#define NIDEVICE_GRPC_DEVICE_ASYNC_SERVER_WRITE_QUEUE_H
+#include <grpcpp/grpcpp.h>
+
+#include <deque>
+#include <mutex>
+#include <queue>
+
+namespace nidevice_grpc {
```

Comment:
Note: I had to add this because the fake test I wrote would crash consistently by queueing up multiple writes. This will also be an issue with other callbacks (but not RegisterDone, which only happens once).